### PR TITLE
fix: panic if daemon command is invalid

### PR DESCRIPTION
**Pull request by kemingy**
Signed-off-by: Keming <kemingyang@tensorchord.ai>

The warning may not be noticed by users.

Let it panic so users know something wrong in the `envd` file.

### CODE REVIEW

**Comment by gaocegege** [comment]
We should use github.com/cockroachdb/errors

**Comment by gaocegege** [comment]
Ref #27

File: pkg/lang/frontend/starlark/runtime/runtime.go [hunk 1]
```diff
@@ -78,8 +78,7 @@ func ruleFuncDaemon(thread *starlark.Thread, _ *starlark.Builtin,
 		for i := 0; i < commands.Len(); i++ {
 			args, ok := commands.Index(i).(*starlark.List)
 			if !ok {
-				logger.Warnf("cannot parse %s into a list of string", commands.Index(i).String())
-				continue
+				return nil, fmt.Errorf("invalid daemon commands (%s)", commands.Index(i).String())
```
Comment (gaocegege) [inline pkg/lang/frontend/starlark/runtime/runtime.go:81]:
```suggestion
				return nil, errors.Newf("invalid daemon commands (%s)", commands.Index(i).String())
```

**Review by gaocegege — APPROVED** [review]

**Comment by gaocegege** [comment]
https://github.com/tensorchord/envd/pull/788/files#diff-826b20bb1048359369852468681128ce52303cb22c3f792476906cc05aabd414R18

Please replace it with github.com/cockroachdb/errors

File: pkg/lang/frontend/starlark/runtime/runtime.go [hunk 2]
```diff
@@ -78,8 +78,7 @@ func ruleFuncDaemon(thread *starlark.Thread, _ *starlark.Builtin,
 		for i := 0; i < commands.Len(); i++ {
 			args, ok := commands.Index(i).(*starlark.List)
 			if !ok {
-				logger.Warnf("cannot parse %s into a list of string", commands.Index(i).String())
-				continue
+				return nil, errors.Newf("invalid daemon commands (%s)", commands.Index(i).String())
```
Comment (zwpaper) [inline pkg/lang/frontend/starlark/runtime/runtime.go:81]: do not forget to change the `import`

**Comment by kemingy** [comment]
There are still 28 `fmt.Errorf` in the codebase, will fix them in another PR.

```console
❯ rrg 'fmt.Errorf' -F -w -t go
pkg/remote/sshd/os.go:26:       errNoShell = fmt.Errorf("failed to find any shell in the PATH")
pkg/ssh/ssh.go:216:             return fmt.Errorf("request for pseudo terminal failed: %w", err)
pkg/ssh/ssh.go:221:             return fmt.Errorf("unable to setup stdin for session: %w", err)
pkg/ssh/ssh.go:227:             return fmt.Errorf("unable to setup stdout for session: %w", err)
pkg/ssh/ssh.go:238:             return fmt.Errorf("unable to setup stderr for session: %w", err)
pkg/ssh/ssh.go:289:                     return nil, fmt.Errorf("decrypting PEM block failed %w", err)
pkg/ssh/ssh.go:301:                     return nil, fmt.Errorf("creating signer from encrypted key failed %w", err)
pkg/ssh/ssh.go:309:                     return nil, fmt.Errorf("parsing plain private key failed %w", err)
pkg/ssh/ssh.go:321:                     return nil, fmt.Errorf("Parsing PKCS private key failed %w", err)
pkg/ssh/ssh.go:328:                     return nil, fmt.Errorf("Parsing EC private key failed %w", err)
pkg/ssh/ssh.go:335:                     return nil, fmt.Errorf("Parsing DSA private key failed %w", err)
pkg/ssh/ssh.go:340:             return nil, fmt.Errorf("Parsing private key failed, unsupported key type %q", block.Type)
pkg/docker/docker.go:111:               return "", fmt.Errorf("invalid repository name (%s), cannot specify 64-byte hexadecimal strings, please rename it", s)
pkg/metrics/collector.go:30:    ErrUnknownCollector := fmt.Errorf("unknown collector: %s", name)
pkg/metrics/collector.go:31:    ErrUnknownHandle := fmt.Errorf("unknown handler: %s", name)
pkg/util/ziputil/unzip.go:105:                  return filenames, fmt.Errorf("%s: illegal file path", fpath)
pkg/lang/ir/util.go:46:                 return "", nil, fmt.Errorf("language %s is not supported", language)
pkg/ssh/config/ssh_config.go:239:                       return fmt.Errorf("failed to get info on %s: %w", p, err)
pkg/ssh/config/ssh_config.go:251:               return fmt.Errorf("failed to create temporary config file: %w", err)
pkg/ssh/config/ssh_config.go:265:               return fmt.Errorf("failed to set permissions to %s: %w", temp.Name(), err)
pkg/ssh/config/ssh_config.go:269:               return fmt.Errorf("new config is not valid: %w", err)
pkg/ssh/config/ssh_config.go:273:               return fmt.Errorf("failed to move %s to %s: %w", temp.Name(), p, err)
pkg/ssh/config/ssh_config.go:462:               return 0, fmt.Errorf("development container not found")
pkg/ssh/config/ssh_config.go:467:               return 0, fmt.Errorf("port not found")
pkg/ssh/config/ssh_config.go:472:               return 0, fmt.Errorf("invalid port: %s", param.value())
pkg/ssh/config/ssh_config.go:526:               return nil, fmt.Errorf("can't open %s: %w", path, err)
pkg/ssh/config/ssh_config.go:533:               return nil, fmt.Errorf("fail to decode %s: %w", path, err)
pkg/ssh/config/ssh_config.go:541:               return fmt.Errorf("fail to update SSH config file %s: %w", path, err)
```

**Review by zwpaper — APPROVED** [review]

**Pull request merged**